Neo4j.Driver/Neo4j.Driver/Internal/ConnectionPool.cs

```diff
@@ -85,12 +80,9 @@ private IPooledConnection CreateNewPooledConnection()
             try
             {
                 conn = _fakeConnection != null
-                    ? new PooledConnection(_fakeConnection, Release)
-                    : new PooledConnection(new SocketConnection(_uri, _connectionSettings, _logger), Release);
-                if (_externalErrorHandler != null)
-                {
-                    conn.ExternalConnectionErrorHander(_externalErrorHandler);
-                }
+                    ? new PooledConnection(()=>_fakeConnection, Release)
```

Comment:
Good idea. I think the initial idea was to avoid using `virtual` modifier. But overall it should make this class easier. I will make this change outside this PR.